shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java

```diff
@@ -1223,11 +1223,31 @@ public static boolean isHdfsEncryptionSupported() {
     private final Configuration conf;
 
     public HdfsEncryptionShim(URI uri, Configuration conf) throws IOException {
-      DistributedFileSystem dfs = (DistributedFileSystem)FileSystem.get(uri, conf);
-
       this.conf = conf;
-      this.keyProvider = dfs.getClient().getKeyProvider();
       this.hdfsAdmin = new HdfsAdmin(uri, conf);
+      this.keyProvider = getKeyProvider();
+    }
+
+    private KeyProvider getKeyProvider() throws IOException {
+      if (isMethodExist(HdfsAdmin.class, "getKeyProvider")) {
```

Comment:
this is not needed. Hive's minimum version required for Hadoop is 3.0

Reply:
That is great. I will remove that then. Thanks for pointing.